db/db_filesnapshot.cc

```diff
@@ -163,22 +163,14 @@ Status DBImpl::GetCurrentWalFile(std::unique_ptr<LogFile>* current_log_file) {
   return wal_manager_.GetLiveWalFile(current_logfile_number, current_log_file);
 }
 
-Status DBImpl::GetLiveFilesStorageInfo(
-    const LiveFilesStorageInfoOptions& opts,
-    std::vector<LiveFileStorageInfo>* files) {
-  // To avoid returning partial results, only move results to files on success.
-  assert(files);
-  files->clear();
-  std::vector<LiveFileStorageInfo> results;
-
-  // NOTE: This implementation was largely migrated from Checkpoint.
-
+Status DBImpl::IsFlushMemtableNeededForGetLiveFiles(
+    const LiveFilesStorageInfoOptions& opts, bool& need_flush_memtable) {
```

Comment:
Not very important, but I think returning a <Status, bool> seems clearer for people reading the code.

Reply:
Will keep as is.